framework/src/plugins/base_plugin.ts

```diff
@@ -252,3 +254,61 @@ export abstract class BasePlugin {
 	public abstract async load(channel: BaseChannel): Promise<void>;
 	public abstract async unload(): Promise<void>;
 }
+
+// TODO: Once the issue fixed we can use require.resolve to rewrite the logic
+//  https://github.com/facebook/jest/issues/9543
+export const getPluginExportPath = (
+	pluginKlass: typeof BasePlugin,
+	strict = true,
+): string | undefined => {
+	let nodeModule: Record<string, unknown> | undefined;
+	let nodeModulePath: string | undefined;
+
+	try {
+		// Check if plugin name is an npm package
+		// eslint-disable-next-line global-require, import/no-dynamic-require, @typescript-eslint/no-var-requires, @typescript-eslint/no-unsafe-assignment
+		nodeModule = require(pluginKlass.info.name);
+		nodeModulePath = pluginKlass.info.name;
+	} catch (error) {
+		/* Plugin info.name is not an npm package */
+	}
+
+	if (!nodeModule && pluginKlass.info.exportPath) {
+		try {
+			// Check if plugin name is an npm package
+			// eslint-disable-next-line global-require, import/no-dynamic-require, @typescript-eslint/no-var-requires, @typescript-eslint/no-unsafe-assignment
+			nodeModule = require(pluginKlass.info.exportPath);
+			nodeModulePath = pluginKlass.info.exportPath;
```

Comment:
same as above, and below condition should be simplified by that

Reply:
Same as above.